### ISSUE TITLE
[bug] user created even when SMTP errors are received

### ISSUE THREAD

**Issue by atb00ker** [openwisp/docker-openwisp#30]
While adding a new user, even if an SMTP error is received, user is still added to the database. Expected response should be `Error code: 5XX`.

Example errors:

- smtplib.SMTPRecipientsRefused: {'controller@mydomain.com': (550, b'5.1.1 <controller@mydomain.com>: Recipient address rejected: User unknown in local recipient table')}
- smtplib.SMTPRecipientsRefused: {'ajay39in@gmail.com': (454, b'4.7.1 <ajay39in@gmail.com>: Relay access denied')}
- socket.gaierror: [Errno -3] Try again

To reproduce:
- Intensionally misconfigure postfix so that it either can't relay(incase relay host is used).
or
- Send response to non-existing users in "mydestinations".
or
- Turn off the postfix service (`postfix stop`)

**Comment by nemesisdesign** [comment]
I don't think this is an issue, if the SMTP doesn't work we should not block the creation of a new user, I think this can be closed :blush: .
